pkg/actors/reminders/reminders.go

```diff
@@ -149,6 +156,11 @@ func (r *reminders) CreateReminder(ctx context.Context, reminder *internal.Remin
 	if err != nil {
 		return fmt.Errorf("error storing reminder: %w", err)
 	}
+	w3cString := diag.SpanContextToW3CString(span.SpanContext())
```

Comment:
Invoking `updateReminderTrack` performs more I/O operations with storage. It will make this a lot slower (than it already is).

Could the tracing stuff be stored with the reminder, rather than with the track, to avoid an additional I/O?

Otherwise, could we update the reminder track only if tracing is enabled for reminders?

Reply:
@DeepanshuA here's an idea, which is similar to how we store the execution time:

- The initial span is stored with the reminder itself
- The track _can_ store a span, which overrides the one from the reminder data if present

This way we avoid adding more storage operations

Reply:
> What I understand right now is that more than number of operations, bigger concern is the data of reminder(s) itself getting stored in key "actors||<actor_type>". If I store traceState in reminder as well, my concern is that reminder data will be more corrupted.

That’s true, but the tracing data is relatively small.

The number of operations is a concern too however, as in many databases it requires an extra round-trip to the DB.